visualization/utils/strategies/base_strategy.py

```diff
@@ -0,0 +1,246 @@
+    ) -> List[dbc.AccordionItem]:
+        switch_layout = [
+            get_switch_layout(
+                {
+                    "type": "view_mode",
+                    "id": QUERY_INPUT_TYPE,
+                },
+                ["view mode"],
+            )
+        ]
+        search_layout = [self._get_search_prompt_layout()] if is_prompt_search else []
+        query_input = [
+            html.Div(
+                self.get_query_input_children_layout(key_values),
+                id="query_input_children",
+            )
+        ]
+        return [
+            dbc.AccordionItem(
+                html.Div(
+                    switch_layout + search_layout + query_input,
+                ),
+                title="Input",
+                id="query_input_content",
+            )
+        ]
+
+    def get_query_input_children_layout(
+        self, key_values: Iterable[Tuple[str, str]], view_mode: bool = False
+    ) -> List[dbc.InputGroup]:
+        return [
+            dbc.InputGroup(
+                [
+                    dbc.InputGroupText(key),
+                    get_text_area_layout(key, value, view_mode),
+                ],
+                className="mb-3",
+            )
+            for key, value in key_values
+        ]
+
+    def get_few_shots_div_layout(self, size: int) -> html.Div:
+        return html.Div(
+            [
+                html.Div(
+                    [
+                        dbc.Pagination(
+                            id="few_shots_pagination",
+                            max_value=size,
+                            active_page=1,
+                        ),
+                        dbc.Button(
+                            "add example",
+                            id="add_example_button",
+                            outline=True,
+                            size="sm",
+                            color="primary",
+                            className="me-1",
+                        ),
+                        dbc.Button(
+                            "delete current example",
+                            id="del_example_button",
+                            outline=True,
+                            size="sm",
+                            color="primary",
+                            className="me-1",
+                        ),
+                        get_switch_layout(
+                            id={
+                                "type": "view_mode",
+                                "id": "few_shots_input",
+                            },
+                            labels=["view mode"],
+                        ),
+                    ]
+                ),
+                dbc.Container(id="few_shots_pagination_content"),
+            ],
+            id="few_shots_div",
+        )
+
+    def run(self, utils: Dict, params: Dict) -> html.Div:
+        self.sandbox_init(utils)
+        llm = get_model(
+            server_type=self.config['server']['server_type'],
```

Comment:
same here, should come directly from the config so that it's possible to customize other potential parameters